TODO.md

```diff
@@ -39,20 +39,10 @@ For more ambitious, long-term features, see [docs/near-future.md](./docs/near-fu
     - Implemented support for interpreting the `slices` and `errors` packages from source by adding language features like 3-index slicing, variadic functions, and support for struct literals with scoped variables.
 - **`docgen` and Symbolic-Execution-like Engine (`symgo`)**: A symbolic execution engine (`symgo`) and a demonstration tool (`docgen`) that uses it to generate OpenAPI 3.1 specifications from `net/http` server code. The engine can trace function calls, resolve types, and evaluate handler logic to infer routes, methods, parameters, and request/response schemas. The `docgen` tool includes a golden-file test suite and can output in both JSON and YAML. ([docs/plan-symbolic-execution-like.md](./docs/plan-symbolic-execution-like.md))
     - Removed manual stubs for standard library types (`net/http`, `io`, etc.) from the `docgen` example, as `go-scan`'s improved module-aware resolution now handles them automatically.
+- **Type-Safe `docgen` Patterns**: Enhanced the `docgen` tool to allow defining analysis patterns using type-safe function references (`Fn: mypkg.MyFunc`) instead of string keys. This involved creating a new `GoSourceFunction` object in the `minigo` interpreter that preserves the function's definition environment, ensuring correct symbol resolution. ([docs/trouble-docgen.md](./docs/trouble-docgen.md))
```

Comment:
```suggestion
- **Type-Safe `docgen` Patterns**: Enhanced the `docgen` tool to allow defining analysis patterns using type-safe function references (`Fn: mypkg.MyFunc`) instead of string keys. This involved creating a new `GoSourceFunction` object in the `minigo` interpreter that preserves the function's definition environment, ensuring correct symbol resolution. ([docs/plan-docgen-minigo-fn-ref.md](./docs/plan-docgen-minigo-fn-ref.md))
```